netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -521,12 +524,54 @@ NettyChannelBuilder setTransportTracerFactory(TransportTracer.Factory transportT
     return this;
   }
 
-  interface ProtocolNegotiatorFactory {
-    /**
-     * Returns a ProtocolNegotatior instance configured for this Builder. This method is called
-     * during {@code ManagedChannelBuilder#build()}.
-     */
-    ProtocolNegotiator buildProtocolNegotiator();
+  @VisibleForTesting
+  @CheckReturnValue
+  static ProtocolNegotiator createProtocolNegotiatorByType(
+      NegotiationType negotiationType,
+      SslContext sslContext,
+      ObjectPool<? extends Executor> executorPool) {
+    switch (negotiationType) {
```

Comment:
Do you want to assert that `sslContext` is `null` if `negotiationType` is `PLAINTEXT` or `PLAINTEXT_UPGRADE`?

Reply:
No, this is pre-existing code, just moved to a less-bad spot. I moved it back to where it was.